File: extract_hubs.js
const fs = require('fs');
const cheerio = require('cheerio');

const html = fs.readFileSync('hubs.html', 'utf-8');
const $ = cheerio.load(html);

const hubs = [];
$('table tr').slice(1).each((i, row) => {
    const cols = $(row).find('td');
    if (cols.length >= 4) {
        hubs.push({
            airline: $(cols[0]).text().trim(),
            iata: $(cols[1]).text().trim(),
            airport: $(cols[2]).text().trim(),
            city: $(cols[3]).text().trim()
        });
    }
});

fs.writeFileSync('hubs.json', JSON.stringify(hubs, null, 2));

File: extract_hubs.py
from bs4 import BeautifulSoup
import json

def extract_hubs():
    with open('hubs.html', 'r', encoding='utf-8') as f:
        soup = BeautifulSoup(f.read(), 'html.parser')
        
    table = soup.find('table')
    if not table:
        return []
    
    hubs = []
    rows = table.find_all('tr')[1:]  # Skip header row
    
    for row in rows:
        cols = row.find_all('td')
        if len(cols) >= 4:
            hub = {
                'airline': cols[0].get_text(strip=True),
                'iata': cols[1].get_text(strip=True),
                'airport': cols[2].get_text(strip=True),
                'city': cols[3].get_text(strip=True)
            }
            hubs.append(hub)
    
    return hubs

if __name__ == '__main__':
    hubs = extract_hubs()
    with open('hubs.json', 'w', encoding='utf-8') as f:
        json.dump(hubs, f, indent=2)
